pkg/cli/add_vcluster_helm.go

```diff
@@ -44,6 +48,46 @@ func AddVClusterHelm(
 		return err
 	}
 
+	// If the vCluster was paused with the helm driver, adding it to the platform will only create the secret for registration
+	// which leads to confusing behavior for the user since they won't see the cluster in the platform UI until it is resumed.
+	if vCluster.Annotations != nil && vCluster.Annotations[constants.PausedAnnotation] == "true" {
```

Comment:
Could you use lifecycle.IsPaused() here? or is there an import cycle / type conflict?